packages/jest-runner/corpus.ts

```diff
@@ -44,16 +55,56 @@ export class Corpus {
 
 	private createMissingDirectories() {
 		fs.mkdirSync(this._inputsDirectory, { recursive: true });
+		fs.mkdirSync(this._corpusDirectory, { recursive: true });
 	}
 }
 
-const buildInputsDirectory = (
+const buildDirectory = (
+	testSourceFilePath: string,
+	testJestPathElements: string[],
+	addToProjectRoot = ""
+): string => {
+	const rootDirectory = buildRootDirectory(
+		testSourceFilePath,
+		addToProjectRoot
+	);
+	const safeTestJestPathElements = testJestPathElements.map(
+		replaceSpacesWithUnderscore
+	);
+	return path.join(rootDirectory, ...safeTestJestPathElements, path.sep);
+};
+
+const buildRootDirectory = (
 	testSourceFilePath: string,
-	testJestPathElements: string[]
+	addToProjectRoot: string
 ): string => {
-	const root = path.parse(testSourceFilePath);
-	const pathElements = testJestPathElements.map(replaceSpacesWithUnderscore);
-	return path.join(root.dir, root.name, ...pathElements) + path.sep;
+	const inputsRoot = path.parse(testSourceFilePath);
+	const testName = inputsRoot.name;
+	let mainDir = inputsRoot.dir;
+	if (addToProjectRoot !== "") {
+		// looking for the root directory of the project
+		mainDir = path.join(
+			findDirectoryWithPackageJson(inputsRoot).dir,
```

Comment:
This would mean that in a project with multiple workspaces (each having its own `package.json`), we would create multiple `.ci-corpus` directories. Is this the expected behavior, or would it be better to have a single generated corpus per project?

Reply:
This is the intended behavior. Merging the corpus directories of multiple workspaces could cause name conflicts. Also, the workspaces are projects in themselfs and hence should have their own test/fuzz/etc. directories. WDYT?